src/Grpc.StatusProto/README.md

```diff
@@ -0,0 +1,256 @@
+    Message = "Simple error message",
+    Details =
+    {
+        Any.Pack(new ErrorInfo
+        {
+            Domain = "error example",
+            Reason = "some reason"
+        }),
+        Any.Pack(new RequestInfo
+        {
+            RequestId = "EchoRequest",
+            ServingData = "Param: " + request.Action.ToString()
+        }),
+        }
+}.ToRpcException();
+```
+Example - setting the status and response trailers instead of throwing an exception:
+```C#
+context.Status = new Grpc.Core.Status(StatusCode.Internal, "Some detail");
+context.ResponseTrailers.SetRpcStatus(new Google.Rpc.Status
+{
+    Code = (int)StatusCode.NotFound,
+    Message = "Simple error message",
+    Details =
+    {
+        Any.Pack(new ErrorInfo
+        {
+            Domain = "error example",
+            Reason = "some reason"
+        })
+    }
+});
+
+return Task.FromResult( /* ... */ );
+```
+
+## Client Side
+
+There is an extension method to retrieve a `Google.Rpc.Status` from the metadata in an `RpcException`.  There are also extension methods to retieve the details from the `Google.Rpc.Status`.
+
+If the client knows what details to expect for a specific error code then it can use the extension methods the explicitly extract the know type from the status details. 
+
+Example:
+```C#
+void PrintError(RpcException ex)
+{
+    // Get the status from the RpcException
+    Google.Rpc.Status? rpcStatus = ex.GetRpcStatus(); // Extension method
+
+    if (rpcStatus != null)
+    {
+        Console.WriteLine($"Google.Rpc Status: Code: {rpcStatus.Code}, Message: {rpcStatus.Message}");
+
+        // Try and get the ErrorInfo from the details
+        ErrorInfo? errorInfo = rpcStatus.GetErrorInfo(); // Extension method
+        if (errorInfo != null)
+        {
+            Console.WriteLine($"\tErrorInfo: Reason: {errorInfo.Reason}, Domain: {errorInfo.Domain}");
+            foreach (var md in errorInfo.Metadata)
+            {
+                Console.WriteLine($"\tKey: {md.Key}, Value: {md.Value}");
+            }
+        }
+        // etc ...
+    }
+}
+```
+
+Alternatively the client can walk the list of details contained in the status, processing
+each object. The class `DetailsTypesRegistry` provides an `Unpack` method to decode the `Any` messages in the status details. C#'s switch pattern matching can be used to process the details.
+
+Example:
+```C#
+void PrintStatusDetails(RpcException ex)
+{
+    // Get the status from the RpcException
+    Google.Rpc.Status? rpcStatus = ex.GetRpcStatus(); // Extension method
+
+    if (rpcStatus != null)
+    {
+        // Decode each "Any" item from the details in turn
+        foreach (Any any in rpcStatus.Details)
+        {
+            // decode the message if it is one of those expected to be in the
+            // status details
+            IMessage msg = DetailsTypesRegistry.Unpack(any); 
+
+            switch (msg)
+            {
+                case null:
```

Comment:
This would go away in UnpackedDetails - we'd just ignore any unknown error details types. That's probably more convenient for users (who can always do it the long way to get all the info).